Implemented Swagger Documentation, Services, Controllers, etc.

jmoran-cei opened the pull request:
Added:
- Trips, Users, Error controllers (includes XML Swagger documentation)
- Trip, User, Destination, and ToDo models
- Trip and User services
- Cross-Origin Policy (Cors)
- Swagger Documentation
- Dependency Injection _[added after code review]_

Update: I've gone through the swagger documentation and resources you sent for creating a web API with .NET. I am going to start learning EF and implementing it into my project. 

Question: I'm assuming so, but is it general standard to provide in-depth XML documentation throughout the entire API (at least in my case I mean to specify documentation for each part of a model, etc.)? I provided documentation for the controllers, but then decided to temporarily suppress the warning for the remaining 35 warnings about ` Missing XML comment for publicly visible type or member)` haha. I'm guessing it's better to be as thorough as can be on a client project, but wasn't sure.

_*Also.. Apologies for the commit cluttering with the removed files! I didn't get one generated initially, so I had to add a .gitignore file to prevent the annoyance of cache files, etc._

jmoran-cei committed:
created models for Destination, Trip, User, and ToDo

jmoran-cei committed:
added User and Trip Services

jmoran-cei committed:
added /Users and /Trips controllers

jmoran-cei committed:
adjusting controllers to  specificy response types for actions, improved readability with shorthand model references, models specify required fields

jmoran-cei committed:
adding gitignore file

jmoran-cei committed:
Generate swagger documentation

jmoran-cei committed:
Removed git ignored files

jmoran-cei committed:
Added swagger documentation for controllers

jmoran-cei committed:
added error controller

jmoran-cei committed:
minor changes

swhook52 commented on Services/UserService.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace TravelTrack_API.Services;
+
+public static class UserService
```
You should take static off of these classes. You'll want to use .NET's built in dependency injection to use the services in your controllers. You'll set that up in your program.cs file. Static is bad for testing and really doesn't gain us anything here.

jmoran-cei replied:
static references removed, dependency injection implemented (not yet with DTOs, context, mapper, etc.)

swhook52 commented on Services/TripService.cs:
```diff
@@ -0,0 +1,95 @@
+
+namespace TravelTrack_API.Services;
+
+public static class TripService
```
Same here, remove static from any services.

jmoran-cei replied:
static references removed, dependency injection implemented (not yet with DTOs, context, mapper, etc.)

swhook52 commented on Controllers/UsersController.cs:
```diff
@@ -0,0 +1,101 @@
+        if (existingUser is null)
+            return NotFound(); // 404
+
+        UserService.Update(user);
```
Typically, your service would return the updated user. Then, you would map down to only properties that the client cares about (think... not returning the password) and return that back to the user as a 200.

That mapping layer is something that you don't have here and that's fine. We do want to add that in the future. Those are called DTOs. They are models specifically made for the client and it is typically a limited set of properties from your actual data or domain models (the representation of the table). The mapping layer can happen in the service layer or the controller layer.

Finally, as I mentioned in the services, you'll want instances of this service injected into the controller constructor instead of static references.

jmoran-cei replied:
static references removed, dependency injection implemented (not yet with DTOs, context, mapper, etc.)

swhook52 commented on Controllers/ErrorController.cs:
```diff
@@ -0,0 +1,27 @@
+/// handles incoming errors (when not in Development)
+///</summary>
+[ApiController]
+public class ErrorController : ControllerBase
```
I don't think you'll need a controller like this for a API. Your errors should be returned in the API call that causes the problem. Maybe you have other plans for this controller though.

jmoran-cei replied:
removed unneeded error controller

jmoran-cei committed:
removed static from all services, created TripService and UserService interfaces (will be more properly implemented in the future with DTOs, context, mappers), removed unneeded errors controller

jmoran-cei commented:
I've completed the following after your review:
- removed all static references
- created interfaces for services (TripService & UserService)
- implemented dependency injection 
- removed unneeded Error Controller

The implementation will be further improved in the future when I get through EF and develop DTOs, mapper, context, etc

jmoran-cei committed:
dependency injection in Program.cs
